### DISCUSSION TITLE
How about lit, or custom web components?

### DISCUSSION THREAD

**Discussion by mohammadaalhomsi** [chakra-ui/ark#2947] [General]
Is there anyway we can use ark as custom web components without any library?

**Comment by cschroeter** [comment]
@mohammadaalhomsi 

It is already a thing. Just need time to do it https://github.com/chakra-ui/zag/tree/main/examples/lit-ts
Feel free to help

**Reply by osdiab** [reply]
Looks like with the release of [Brisa](https://brisa.build/) that framework would benefit greatly from this as well.

**Reply by osdiab** [reply]
This is how you configure them there:

- [Docs](https://brisa.build/building-your-application/components-details/external-libraries#web-components-dependencies)
- [Example](https://github.com/brisa-build/brisa/tree/main/examples/with-external-web-component)

Seems like having modules that export the ready-to-use web components and typings would be useful for that one, but idk what would go into that.

**Comment by mohammadaalhomsi** [comment]
Thank you for your reply, I went quickly through the example, but I see it's really tide with your internal API for state management and property updates.

In my projects, I try to leverage the native browser APIs to the maximum. Even lit functions, I merly use `requestUpdate` and that's it.

I used 10 years old library for the tags input and integrating it within a simple lit element was half an hour task. https://github.com/pirxpilot/tags-input
